File: src/Sample/Flutter.Net/ListViewSeparatedDemoPage.cs
using Avalonia;
using Avalonia.Media;
using Flutter.Foundation;
using Flutter.Rendering;
using Flutter.Widgets;

// Dart parity source (reference): dart_sample/lib/list_view_separated_demo_page.dart (exact sample parity)

namespace Flutter.Net;

public sealed class ListViewSeparatedDemoPage : StatelessWidget
{
    public override Widget Build(BuildContext context)
    {
        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 10,
            children:
            [
                new Text("ListView.Separated", fontSize: 20, color: Colors.Black),
                new Text(
                    "Uses separate builders for rows and separators.",
                    fontSize: 14,
                    color: Colors.DimGray),
                new Expanded(
                    child: ListView.Separated(
                        itemCount: 30,
                        padding: new Thickness(12),
                        itemBuilder: (_, index) => new SeparatedListItem(index),
                        separatorBuilder: (_, index) => new Container(
                            height: 4,
                            color: index % 2 == 0
                                ? Color.Parse("#FFDCE3ED")
                                : Color.Parse("#FFE9EEF5")),
                        addAutomaticKeepAlives: false)),
            ]);
    }
}

internal sealed class SeparatedListItem : StatefulWidget
{
    public SeparatedListItem(int index, Key? key = null) : base(key)
    {
        Index = index;
    }

    public int Index { get; }

    public override State CreateState()
    {
        return new SeparatedListItemState(Index);
    }
}

internal sealed class SeparatedListItemState : State
{
    private readonly int _index;
    private int _taps;

    public SeparatedListItemState(int index)
    {
        _index = index;
    }

    public override Widget Build(BuildContext context)
    {
        return new CounterTapButton(
            label: $"item #{_index} taps={_taps}",
            onTap: () => SetState(() => _taps += 1),
            background: _index % 2 == 0 ? Color.Parse("#FFEFF5FF") : Colors.White,
            foreground: Colors.Black,
            fontSize: 13,
            padding: new Thickness(10, 8));
    }
}

File: dart_sample/lib/list_view_separated_demo_page.dart
import 'package:flutter/material.dart';

import 'counter_widgets.dart';

class ListViewSeparatedDemoPage extends StatelessWidget {
  const ListViewSeparatedDemoPage({super.key});

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 10,
      children: <Widget>[
        const Text(
          'ListView.Separated',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'Uses separate builders for rows and separators.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Expanded(
          child: ListView.separated(
            itemCount: 30,
            padding: const EdgeInsets.all(12),
            itemBuilder: (BuildContext context, int index) {
              return SeparatedListItem(index: index);
            },
            separatorBuilder: (BuildContext context, int index) {
              return Container(
                height: 4,
                color: index.isEven
                    ? const Color(0xFFDCE3ED)
                    : const Color(0xFFE9EEF5),
              );
            },
          ),
        ),
      ],
    );
  }
}

class SeparatedListItem extends StatefulWidget {
  const SeparatedListItem({required this.index, super.key});

  final int index;

  @override
  State<SeparatedListItem> createState() => _SeparatedListItemState();
}

class _SeparatedListItemState extends State<SeparatedListItem> {
  int _taps = 0;

  @override
  Widget build(BuildContext context) {
    return CounterTapButton(
      label: 'item #${widget.index} taps=$_taps',
      onTap: () => setState(() => _taps += 1),
      background: widget.index.isEven ? const Color(0xFFEFF5FF) : Colors.white,
      foreground: Colors.black,
      fontSize: 13,
      padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 8),
    );
  }
}
